pkg/agent/agentserver/server.go

```diff
@@ -453,38 +435,45 @@ func (s *ProxyServer) serveRecvBackend(stream agent.AgentService_ConnectServer,
 					klog.Warningf("<<< DIAL_RSP send to client stream error: %v", err)
 				} else {
 					client.connectID = resp.ConnectID
-					s.Frontends[resp.ConnectID] = client
+					s.addFrontend(agentID, resp.ConnectID, client)
 					close(client.connected)
 				}
 			}
 
 		case client.PacketType_DATA:
 			resp := pkt.GetData()
 			klog.Infof("<<< Received DATA(id=%d)", resp.ConnectID)
-			if client, ok := s.Frontends[resp.ConnectID]; ok {
-				if err := client.send(pkt); err != nil {
-					klog.Warningf("<<< DATA send to client stream error: %v", err)
-				} else {
-					klog.Infof("<<< DATA sent to frontend")
-				}
+			client, err := s.getFrontend(agentID, resp.ConnectID)
+			if err != nil {
+				klog.Warning(err)
+				break
+			}
+			if err := client.send(pkt); err != nil {
+				klog.Warningf("<<< DATA send to client stream error: %v", err)
+			} else {
+				klog.V(6).Infof("<<< DATA sent to frontend")
 			}
 
 		case client.PacketType_CLOSE_RSP:
 			resp := pkt.GetCloseResponse()
 			klog.Infof("<<< Received CLOSE_RSP(id=%d)", resp.ConnectID)
-			if client, ok := s.Frontends[resp.ConnectID]; ok {
-				if err := client.send(pkt); err != nil {
-					// Normal when frontend closes it.
-					klog.Warningf("<<< CLOSE_RSP send to client stream error: %v", err)
-				} else {
-					klog.Infof("<<< CLOSE_RSP sent to frontend")
-				}
+			client, err := s.getFrontend(agentID, resp.ConnectID)
+			if err != nil {
+				klog.Warning(err)
+				break
 			}
+			if err := client.send(pkt); err != nil {
+				// Normal when frontend closes it.
+				klog.Warningf("<<< CLOSE_RSP send to client stream error: %v", err)
```

Comment:
I agree that this would be normal but then we should lower the log level. Info and at least 4.